sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -1869,107 +1912,235 @@ class CompactBlockProcessor internal constructor(
                 emit(SyncingResult.EnhanceSuccess)
             }
 
-        private suspend fun enhanceTransaction(
-            transaction: DbTransactionOverview,
+        private suspend fun processTransparentAddressTxids(
+            transactionRequest: TransactionDataRequest.SpendsFromAddress,
             backend: TypesafeBackend,
-            downloader: CompactBlockDownloader
+            downloader: CompactBlockDownloader,
+            network: ZcashNetwork
         ): SyncingResult {
-            Twig.debug {
-                "Starting enhancing transaction (txid:${transaction.txIdString()}  block:${transaction
-                    .minedHeight})"
-            }
-            if (transaction.minedHeight == null) {
-                return SyncingResult.EnhanceSuccess
+            Twig.debug { "Starting to get transparent address transactions ids" }
+
+            val traceScope = TraceScope("CompactBlockProcessor.processTransparentAddressTxids")
+            val result =
+                try {
+                    // Fetching transactions is done with retries to eliminate a bad network condition
+                    val listOfResults =
+                        getTransparentAddressTransactions(
+                            transactionRequest = transactionRequest,
+                            downloader = downloader,
+                        )
+
+                    Twig.debug { "Obtained ${listOfResults.size} transactions for processing" }
+
+                    listOfResults.forEach { rawTransactionUnsafe ->
+                        when (rawTransactionUnsafe) {
+                            is RawTransactionUnsafe.MainChain -> {
+                                // Decrypting and storing transaction is run just once, since we consider it more stable
+                                Twig.verbose {
+                                    "Decrypting and storing transaction at height: ${rawTransactionUnsafe.height.value}"
+                                }
+                                decryptTransaction(
+                                    rawTransaction =
+                                    RawTransaction.new(
+                                        rawTransactionUnsafe = rawTransactionUnsafe,
+                                        network = network
+                                    ),
+                                    backend = backend
+                                )
+                            }
+                            is RawTransactionUnsafe.Mempool,
+                            is RawTransactionUnsafe.OrphanedBlock -> {
+                                error("Unsupported transaction type: $rawTransactionUnsafe")
```

Comment:
This is the second thing I'm still unsure of. Should we treat these cases the same as we do in the case of the `enhanceTransaction` function from the line 2050, or is this fine?

Reply:
Fixed by treating RawTransactionUnsafe the same in case of the SpendsFromAddress flow